tasks.py

```diff
@@ -157,13 +161,27 @@ def automerge(c: Context):
 
 
 @task(aliases=("vuln",))
-def vulnerability_scan(c: Context):
+def vulnerability_scan(c: Context, modified_files_only: bool = False):
     requirements_files = Path().parent.glob("*requirements.txt")
-    for requirements_file in requirements_files:
-        c.run(
-            f"snyk test --file={requirements_file} --package-manager=pip",
-            pty=NOT_WINDOWS,
-        )
+
+    if modified_files_only:
+        if filetype_modified_since_main(c, re.compile(r"requirements\.txt$")):
+            for requirements_file in requirements_files:
+                c.run(
+                    f"snyk test --file={requirements_file} --package-manager=pip",
+                    pty=NOT_WINDOWS,
+                )
+        else:
+            print(
+                "🟢 No requirements.txt files modified since main, skipping vulnerability scan",
+            )
+
+    else:
+        for requirements_file in requirements_files:
+            c.run(
+                f"snyk test --file={requirements_file} --package-manager=pip",
+                pty=NOT_WINDOWS,
+            )
```

Comment:
Can be implemented as a guard, decreasing conditionality and removing the else-statement:

```suggestion
        if not filetype_modified_since_main(c, re.compile(r"requirements\.txt$")):
            print(
                "🟢 No requirements.txt files modified since main, skipping vulnerability scan",
            )
            return None
            
        for requirements_file in requirements_files:
            c.run(
                f"snyk test --file={requirements_file} --package-manager=pip",
                pty=NOT_WINDOWS,
            )
```